l10n_de_tax_statement/models/l10n_de_tax_statement_line.py

```diff
@@ -0,0 +1,174 @@
+            else:
+                line.is_readonly = True
+
+    @api.multi
+    def unlink(self):
+        for line in self:
+            if line.statement_id.state == 'posted':
+                raise UserError(
+                    _('You cannot delete lines of a posted statement! '
+                      'Reset the statement to draft first.'))
+        super(VatStatementLine, self).unlink()
+
+    @api.multi
+    def view_tax_lines(self):
+        self.ensure_one()
+        return self.get_lines_action(tax_or_base='tax')
+
+    @api.multi
+    def view_base_lines(self):
+        self.ensure_one()
+        return self.get_lines_action(tax_or_base='base')
+
+    def get_lines_action(self, tax_or_base='tax'):
+        self.ensure_one()
+        action = self.env.ref('account.action_account_moves_all_tree')
+        vals = action.read()[0]
+        vals['context'] = {}
+        vals['domain'] = self._get_move_lines_domain(tax_or_base)
+        return vals
+
+    def _get_move_lines_domain(self, tax_or_base):
+        statement = self.statement_id
+        taxes = self._filter_taxes_by_code(statement._compute_taxes())
+        past_taxes = statement._compute_past_invoices_taxes()
+        past_taxes = self._filter_taxes_by_code(past_taxes)
+        if statement.state == 'draft':
+            domain = self._get_domain_draft(taxes, tax_or_base)
+            past_domain = self._get_domain_draft(past_taxes, tax_or_base)
+        else:
+            domain = self._get_domain_posted(taxes, tax_or_base)
+            past_domain = self._get_domain_posted(past_taxes, tax_or_base)
+        curr_amls = self.env['account.move.line'].search(domain)
+        past_amls = self.env['account.move.line'].search(past_domain)
+        res = [('id', 'in', past_amls.ids + curr_amls.ids)]
+        return res
+
+    def _filter_taxes_by_code(self, taxes):
+        self.ensure_one()
+        tags_map = self.statement_id._get_tags_map()
+        filtered_taxes = self.env['account.tax']
+        for tax in taxes:
+            for tag in tax.tag_ids:
+                tag_map = tags_map.get(tag.id)
+                if tag_map and tag_map[0] == self.code:
+                    filtered_taxes |= tax
+        return filtered_taxes.with_context(taxes.env.context)
+
+    def _get_domain_draft(self, taxes, tax_or_base):
+        self.ensure_one()
+        ctx = taxes.env.context.copy()
+        ctx.update({
+            'l10n_de_statement_tax_ids': taxes.ids
+        })
+        AccountTax = self.env['account.tax'].with_context(ctx)
+        return AccountTax.get_move_lines_domain(tax_or_base=tax_or_base)
+
+    def _get_domain_posted(self, taxes, tax_or_base):
+        self.ensure_one()
+        statement = self.statement_id
+        domain = [('move_id.l10n_de_tax_statement_id', '=', statement.id)]
+        if tax_or_base == 'tax':
+            tax_domain = [('tax_line_id', '=', taxes.ids)]
+        else:
+            tax_domain = [('tax_ids', '=', taxes.ids)]
```

Comment:
```suggestion
            tax_domain = [('tax_ids', 'in', taxes.ids)]
```

Reply:
I have applied your proposed change. Thanks a lot for your intervention!